pkg/daemon/daemon.go

```diff
@@ -124,6 +124,11 @@ type Daemon struct {
 	drainer *drain.Helper
 }
 
+// CoreOSDaemon protects the methods that should only be called on CoreOS variants
+type CoreOSDaemon struct {
```

Comment:
Ah, interesting approach to add type safety!  Very nice.  I live in Rust and C/C++ land too much that I forget about this Go idiom.